crates/astria-conductor/src/executor/mod.rs

```diff
@@ -396,6 +412,35 @@ impl Executor {
         // TODO(https://github.com/astriaorg/astria/issues/624): add retry logic before failing hard.
```

Comment:
Comment for above: the error cases are not actual errors but part of normal execution. As such, I would change the return type of `execute_soft` to encode if a restart should happen due to the max being reached.